spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -289,20 +297,43 @@ object DistriOptimizer {
         numFinishedModelUpdates >= driverSubModelNum * (1.0 - maxDropPercentage)) {
         // enough records were processed for this batch, so update the model
         val value = lossSum.value / numFinishedModelUpdates
+
+        var l2Norm = 0.0f
+        var scale = ev.fromType(numFinishedModelUpdates)
+        if (normClippingEnable) {
```

Comment:
If the model (e.g., AlexNet or VGG) has large parameter sizes, the overhead can be large? Shall we do this using multi-threading?

Reply:
Updated